app/models/talk.rb

```diff
@@ -111,6 +112,6 @@ def talk_path
     end
 
     def remove_duplicate_dates
-      update_columns(dates: dates.uniq)
+      update_columns(dates: dates.uniq) # rubocop:todo Rails/SkipsModelValidations
     end
```

Comment:
How do we want to address this? Does `update_columns` refer to the currently dealt-with Talk or to all Talks in the database since it's not prefixes with something like `my_talk.`

Reply:
It refers to the currently dealt-with talk (it is equivalent to` self.update_columns(dates: date.uniq)` for which Rubocop would tell you to omit  the `self`). Note that it is not possible to replace `update_column` by `update` here: this would trigger an infinite loop since `update` would again call the `after_save` action which would then call `update` and so on.

However, what we could do here is the following: Replace the line

```ruby
 after_save :remove_duplicate_dates
```

by 

```ruby
before_save :remove_duplicate_dates
```

and replace the `remove_duplicate_dates` method by 

```ruby
def remove_duplicate_dates
  dates.uniq!
end
```

You should try it out e.g. in `/talks/1/edit`.

Reply:
Fixed in eb92c86.

Thanks, I've tried it out and it works fine. Note that a more elegant solution would probably be to have a **set instead of an array** as underlying data structure in order to avoid duplicates. I've added a todo note as comment; this will presumably involve a migration to change the data type on the existing data.

Reply:
I am not sure that set is supported by the adapter for ActiveRecord, see [here](https://guides.rubyonrails.org/active_record_postgresql.html) for the ist of supported data types.

Reply:
Ah, it's a pity PostgreSQL doesn't support sets natively. Then, we'll have to live with it ;)